### ISSUE TITLE
Why c.div_(self.args.batch_size) is needed?

### ISSUE THREAD

**Issue by mongeoroo** [facebookresearch/barlowtwins#56]
Hello, I wanna ask about the division.

> import torch
> import torch.nn.functional as F
> x = torch.rand(8,128)
> y = torch.rand(8,128)
> x = F.normalize(x, dim=0)
> y = F.normalize(y, dim=0)
> torch.allclose(x[:4].T @ y[:4] + x[4:].T @ y[4:], x.T @ y)
> True

According to the code above, we don't need to divide by batch size to combine the vectors distributed in multi-gpus.

Why do we have to divide the cross correlation matrix by batch size?

**Comment by mongeoroo** [comment]
It comes from the difference between normalization and standardization.

The batchnorm is standardization operation. And the L2 norm of the standardization is root of N.

the cross-correlation equation have division by L2 norms of the two variables. Then we should divide it by N.
